packages/nx/src/native/project_graph/types.rs

```diff
@@ -0,0 +1,115 @@
+use napi::bindgen_prelude::Either8;
+use napi::Either;
+use std::collections::HashMap;
+
+#[napi(object)]
+pub struct ExternalNodeData {
+    pub version: String,
+    pub hash: String,
+}
+
+#[napi(object)]
+pub struct ExternalNode {
+    pub version: String,
+    pub hash: String,
+}
+
+// type ExpandedSelfInput =
+// | { fileset: string }
+// | { runtime: string }
+// | { env: string }
+// | { externalDependencies: string[] };
+//
+// type ExpandedDepsOutput = {
+// dependentTasksOutputFiles: string;
+// transitive?: boolean;
+// };
+//
+// type ExpandedInput = ExpandedSelfInput | ExpandedDepsOutput;
```

Comment:
I know this was used as a placeholder to do the conversion to rust, I can remove it later, or you can do it now 😄